scipy/integrate/_tanhsinh.py

```diff
@@ -1265,7 +1242,10 @@ def _integral_bound(f, a, b, step, args, constants):
     # Get a lower bound on the sum and compute effective absolute tolerance
     lb = _tanhsinh(f, a, b, args=args, atol=atol, rtol=rtol, log=log)
     tol = np.broadcast_to(atol, lb.integral.shape)
-    tol = _logsumexp((tol, rtol + lb.integral)) if log else tol + rtol*lb.integral
+    if log:
+        tol = special.logsumexp((tol, rtol + lb.integral), axis=0)
+    else:
+        tol = tol + rtol*lb.integral
```

Comment:
@mdhaber I'm not sure whether this `axis=0` is supposed to be needed or not - there are failures for `axis=None` but everything passes locally with `axis=0` here.

Reply:
I took a look. It already does pass the desired axis everywhere except here, and the desired axis here would be `axis=0` because including the tolerances in a list effectively stacks them along a new zeroth axis and we are trying to add them.